### PR TITLE
Configure repository metadata

### LINKED ISSUE #271: Configure repository metadata
We can configure this repository's metadata by `.asf.yaml`. See also: https://cwiki.apache.org/confluence/display/INFRA/Git+-+.asf.yaml+features

We should change at least notification configurations. All issue changes will be notified to dev@arrow.apache.org with the default configuration.

### PR DESCRIPTION
**Pull request by kou**
fix #271

### CODE REVIEW

File: .asf.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  homepage: https://arrow.apache.org/
+  labels:
+    - apache-arrow
+    - julia
+  features:
+    issues: true
+  enabled_merge_buttons:
+    squash: true
+    merge: false
+    rebase: false
```
Comment (kou) [inline .asf.yaml:36]: @quinnj It seems that JuliaData/Arrow.jl uses only squash merge.

File: .asf.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  protected_branches:
+    main:
+      required_linear_history: true
+      required_pull_request_reviews:
+        dismiss_stale_reviews: true
+        required_approving_review_count: 1
```
Comment (kou) [inline .asf.yaml:42]: @quinnj It seems that JuliaData/Arrow.jl uses protected branch feature but I don't know the configuration. Are there some missing configurations?

See also https://docs.github.com/en/rest/reference/branches#update-branch-protection for available configurations.

File: .asf.yaml (hunk 1 above)
Comment (quinnj) [reply .asf.yaml:36]: We allowed squash and rebase, but not merge.

File: .asf.yaml (hunk 2 above)
Comment (quinnj) [reply .asf.yaml:42]: We also had "Automatically delete head branches " feature enabled, but I think that's the only one I don't see in the list here.

File: .asf.yaml (hunk 1 above)
Comment (kou) [reply .asf.yaml]: OK. I've enabled rebase.

File: .asf.yaml (hunk 2 above)
Comment (kou) [reply .asf.yaml:42]: Thanks.
It seems that we can enable the feature by https://docs.github.com/en/rest/reference/repos#update-a-repository API.
But the `.asf.yaml` mechanism doesn't support it yet: https://github.com/apache/infrastructure-p6/blob/production/modules/gitbox/files/asfgit/package/asfyaml.py#L234
It seems that the `.asf.yaml` mechanism doesn't change the current configuration. I hope that we can still use the feature. We can confirm it by merging this pull request.

File: .asf.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+# https://cwiki.apache.org/confluence/display/INFRA/Git+-+.asf.yaml+features
+
+notifications:
+  commits: commits@arrow.apache.org
+  issues: issues@arrow.apache.org
+  pullrequests: github@arrow.apache.org
+
+github:
+  description: "Apache Arrow Julia"
```
Comment (pitrou) [inline .asf.yaml:26]:
```suggestion
  description: "Official Julia implementation of Apache Arrow"
```

**Comment by pitrou** [comment]
Can we get this merged, at least the notifications part? dev@arrow.apache.org is getting notifications for each PR comment...

**Comment by kou** [comment]
@pitrou Could you "approve" this pull request? We can't merge a pull request without at least one "approve" in this repository.

**Review by pitrou — APPROVED** [review]

**Comment by kou** [comment]
@quinnj I'll merge this pull request to change notifications to `github@arrow.apache.org` from `dev@arrow.apache.org`. We can work on extra configurations in follow-up pull requests. Please comment missing configurations to #271.

**Comment by kou** [comment]
Ah, we can't test "Automatically delete head branches" feature with this pull request because I created a branch for this pull request on my fork.

**Pull request merged**